datalad/tests/utils.py

```diff
@@ -231,6 +235,14 @@ def newfunc(*arg, **kw):
                 # well -- if something wrong just proceed with defaults
                 pass
 
+        # if DATALAD_TESTS_TEMPDIR is set, use that as directory,
+        # let mktemp handle it otherwise. However, an explicitly provided
+        # dir=... will override this.
+        directory = os.environ.get('DATALAD_TESTS_TEMPDIR')
```

Comment:
Nice.
This reminds me of something: My IDE is complaining that "mock" is not listed in requirements.
So, if want full selftesting we may be should include it.